apps/rabbitmq.go

```diff
@@ -16,9 +16,92 @@ package apps
 
 import (
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
 )
 
+type LoggingProcessorRabbitmq struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+}
+
+func (*LoggingProcessorRabbitmq) Type() string {
+	return "rabbitmq"
+}
+
+func (p *LoggingProcessorRabbitmq) Components(tag, uid string) []fluentbit.Component {
+	c := []fluentbit.Component{}
+	regexParser := confgenerator.LoggingProcessorParseRegex{
+		// Sample log line:
+		// 2022-01-31 18:01:20.441571+00:00 [erro] <0.692.0> ** Connection attempt from node 'rabbit_ctl_17@keith-testing-rabbitmq' rejected. Invalid challenge reply. **
+		Regex: `^(?<timestamp>\d+-\d+-\d+ \d+:\d+:\d+\.\d+\+\d+:\d+) \[(?<severity>\w+)\] \<(?<process_id>\d+\.\d+\.\d+)\> (?<message>.*)$`,
+		ParserShared: confgenerator.ParserShared{
+			TimeKey:    "timestamp",
+			TimeFormat: "%Y-%m-%d %H:%M:%S.%L+%Z",
+		},
+	}
+	c = append(c, regexParser.Components(tag, uid)...)
+
+	// severities documented here: https://www.rabbitmq.com/logging.html#log-levels
+	c = append(c, fluentbit.TranslationComponents(tag, "severity", "logging.googleapis.com/severity", true, []struct {
+		SrcVal  string
+		DestVal string
+	}{
+		{
+			"debug", "DEBUG",
+		},
+		{
+			"error", "ERROR",
+		},
+		{
+			"info", "INFO",
+		},
+		{
+			"noti", "DEFAULT",
+		},
```

Comment:
[nit][formatting] For consistency with the other receivers, could we have all of the pairings be inline with their braces like:

```
c = append(c, fluentbit.TranslationComponents(tag, "severity", "logging.googleapis.com/severity", true, []struct {
		SrcVal  string
		DestVal string
	}{
		{"debug", "DEBUG"},
		{"error", "ERROR"},
		{"info", "INFO"},
		{"noti", "DEFAULT"},
	}...,
)
```